processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -565,6 +566,32 @@ public Integer getDimensionIndex(String dimension)
     return dimensionOrder.get(dimension);
   }
 
+  public LinkedHashMap<String, Integer> getDimensionOrder()
+  {
+    return dimensionOrder;
+  }
+
+  public void loadDimensionOrder(LinkedHashMap<String, Integer> oldDimensionOrder)
+  {
+    dimensionOrder.putAll(oldDimensionOrder);
+    for (String dim : oldDimensionOrder.keySet()) {
+      if (dimValues.get(dim) == null) {
+        dimensions.add(dim);
+        dimValues.add(dim);
+      }
+    }
+  }
+
+  public void loadDimensionIterable(Iterable<String> oldDimensionOrder)
+  {
+    int i = 0;
+    for (String dim : oldDimensionOrder) {
+      dimensionOrder.put(dim, i);
+      dimensions.add(dim);
```

Comment:
@nishantmonu51 I think it's necessary, dimValues will throw an exception if you add something that already exists, and dimensions is an ArrayList (don't want duplicates)